algebird-core/src/main/scala/com/twitter/algebird/CountMinSketch.scala

```diff
@@ -812,18 +827,44 @@ case class HeavyHitter[K](item: K, count: Long) extends java.io.Serializable
 
 object HeavyHitter {
 
-  def ordering[K: Ordering]: Ordering[HeavyHitter[K]] = Ordering.by { hh: HeavyHitter[K] => (hh.count, hh.item) }
+  import scala.reflect._
+
+  // Note that we can't define the ordering on `count` only.  We use a set to track heavy hitters, where equality means
+  // two elements are identical.  If we were to use `count` only, we would label different items in heavy hitters as
+  // identical simply because they have the same count (and such false duplicates may happen easily in practice).  For
+  // this reason we need to add another dimension to the ordering to prevent such false duplicates.  For this second
+  // dimension we use an item's default Java hash code because we do not want to introduce an `Ordering` constraint.
+  def ordering[K: ClassTag]: Ordering[HeavyHitter[K]] = {
```

Comment:
damn scala. Equiv and scala.util.hashing.Hashing should be what we want here, but they are so broken and impoverished I guess we should think about it.

I really don't like using ClassTag in general, and I especially don't like it here when it is in fact a round-about way to get at what we really want, which is a hashcode.

So, I guess the least ugly solution is to keep the `Ordering[K]` bound on cases with the TopK heavy hitters. What do you think?